core/trino-main/src/main/java/io/trino/event/SplitMonitorConfig.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.event;
+
+import io.airlift.configuration.Config;
+import io.airlift.units.DataSize;
+import io.airlift.units.DataSize.Unit;
+import io.airlift.units.MaxDataSize;
+import io.airlift.units.MinDataSize;
+
+import javax.validation.constraints.NotNull;
+
+public class SplitMonitorConfig
+{
+    private DataSize maxSplitOutputStageJsonSize = DataSize.of(16, Unit.MEGABYTE);
```

Comment:
I like this default because it's consistent with QueryMonitorConfig. No changes requested here from my side. cc @raunaqmorarka if you have an opinion, given that this is putting a default cap on previously unbounded value.